elytron/WFLY-14984-encrypted-expression-for-client-config.adoc

```diff
@@ -0,0 +1,164 @@
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The ability to specify attribute values as encrypted expressions should be added to wildfly-config.xml. 
+
+* The parsing needs to be able to run separately from the application server, as the server may not be running while the client configuration is being set up. 
+
+* Configurations may not depend on components created for server configuration, such as `resolvers` or `elytron/expression`. The resolver for the encrypted expression should be configured through the client configuration. 
+
+* The `encrypted-expression` configuration can be specified as follows: 
+```
+<encrypted-expression>
+	<credential-store name="credential-store-name" type="PKCS12" provider="provider-name"/>
+        <attributes>
+            <attribute name="path" value="path/to/credential-store-name.cs" />
+        </attributes>
+        <protection-parameter-credentials>
+            <clear-password password="password" />
+        </protection-parameter-credentials>
+            </credential-store>
+	<resolver name= “resolver-name” credential-store= “credential-store-name key=“key”/>
+</encrypted-expression>
+```
+Note that the initial password for the secret-key-credential-store used for the resolver would need to be specified in clear-text. 
+
+Once parsing is done, the resolver would need to be created programmatically (if not created by the server already) and be connected with the secret-key-credential-store. 
+
+* Once this has been configured, secret keys can be added to the credential store under different aliases and be used for client configuration as follows: 
+```
+<credential-store name="credential-store-name" type="PKCS12" provider="provider-name" >
+    <attributes>
+        <attribute name="path" value="path/to/credential-store-name.cs" />
+    </attributes>
+    <protection-parameter-credentials>
+	    <clear-password password={ENC::resolver-name:encrypted-expression} />
+    </protection-parameter-credentials>
+</credential-store>
+```
+
+* The encrypted expression would be created using the `--generate-secret-key` command and can be read using the `--export-secret-key` command. 
+
+=== Nice-to-Have Requirements
+
+* The parsing of the encrypted expression could be added as a separate namespace in the schema and should be added at the top of the xml file. 
+
+** While parsing the xml file, the section for encrypted expression and its resolver would need to be parsed before any other sections that reference it. 
+
+* Encrypted expressions can be added to other subsystems in addition to elytron. As a result, the parsing rules and sequences should be configured to be compatible with those subsystems as well. 
```

Comment:
Since this is client side, the terminology here needs to be updated. It's currently referencing subsystems.

Reply:
fixed!